Panic with nil argument

Hi ,
I found the following panic:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x65d0a5]

goroutine 59 [running]:
testing.tRunner.func1.1(0x1bad020, 0x2ea23f0)
	/usr/local/go/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc0000c8900)
	/usr/local/go/src/testing/testing.go:1075 +0x41a
panic(0x1bad020, 0x2ea23f0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/cenkalti/rpc2/jsonrpc.(*jsonCodec).WriteRequest(0xc00043b4a0, 0xc00013e780, 0x0, 0x0, 0xc0008af388, 0x18)
	/go/pkg/mod/github.com/cenkalti/rpc2@v0.0.0-20210206021708-de76ddb08fa8/jsonrpc/jsonrpc.go:181 +0x85
github.com/cenkalti/rpc2.(*Client).send(0xc00013e770, 0xc00002e690)
	/go/pkg/mod/github.com/cenkalti/rpc2@v0.0.0-20210206021708-de76ddb08fa8/client.go:326 +0x1bf
github.com/cenkalti/rpc2.(*Client).Go(0xc00013e770, 0x1e6de70, 0x1c, 0x0, 0x0, 0x1a89540, 0xc0004c7a60, 0xc000700540, 0xc00002e640)
	/go/pkg/mod/github.com/cenkalti/rpc2@v0.0.0-20210206021708-de76ddb08fa8/client.go:265 +0xd0
github.com/cenkalti/rpc2.(*Client).Call(...)
	/go/pkg/mod/github.com/cenkalti/rpc2@v0.0.0-20210206021708-de76ddb08fa8/client.go:271
```
When calling a client with nil args like so:
```
client.Call("ping", nil,&reply)
```
This worked before this [PR](https://github.com/cenkalti/rpc2/pull/33) was merged.

Thanks,
Trial97

--- a/jsonrpc/jsonrpc.go
+++ b/jsonrpc/jsonrpc.go
@@ -84,9 +84,9 @@ type serverResponse struct {
 	Error  interface{}      `json:"error"`
 }
 type clientRequest struct {
-	Method string        `json:"method"`
-	Params interface{}   `json:"params"`
-	Id     *uint64       `json:"id"`
+	Method string      `json:"method"`
+	Params interface{} `json:"params"`
+	Id     *uint64     `json:"id"`
 }
 
 func (c *jsonCodec) ReadHeader(req *rpc2.Request, resp *rpc2.Response) error {
@@ -178,7 +178,7 @@ func (c *jsonCodec) WriteRequest(r *rpc2.Request, param interface{}) error {
 	req := &clientRequest{Method: r.Method}
 
 	// Check if param is a slice of any kind
-	if reflect.TypeOf(param).Kind() == reflect.Slice {
+	if param != nil && reflect.TypeOf(param).Kind() == reflect.Slice {
 		// If it's a slice, leave as is
 		req.Params = param
 	} else {

--- a/jsonrpc/jsonrpc_test.go
+++ b/jsonrpc/jsonrpc_test.go
@@ -58,6 +58,14 @@ func TestJSONRPC(t *testing.T) {
 		}
 		return nil
 	})
+	srv.Handle("nilArgs", func(client *rpc2.Client, args []interface{}, reply *[]string) error {
+		for _, v := range args {
+			if v == nil {
+				*reply = append(*reply, "nil")
+			}
+		}
+		return nil
+	})
 	number := make(chan int, 1)
 	srv.Handle("set", func(client *rpc2.Client, i int, _ *struct{}) error {
 		number <- i
@@ -154,6 +162,17 @@ func TestJSONRPC(t *testing.T) {
 	expected = []string{"1", "2"}
 	typedArgs := []int{1, 2}
 	err = clt.Call("typedArgs", typedArgs, &reply)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = testArgs(expected, reply); err != nil {
+		t.Fatal(err)
+	}
+
+	// Test nil args
+	reply = []string{}
+	expected = []string{"nil"}
+	err = clt.Call("nilArgs", nil, &reply)
 	if err != nil {
 		t.Fatal(err)
 	}

